gadgets/cmr/CmrRealTimeLAXCineAIAnalysisGadget.cpp

```diff
@@ -15,6 +14,77 @@
 
 namespace Gadgetron { 
 
+    // a singleton class to load model
+    // the current finding is the python runtime may not always release memory after loading in the model and related libraries
+    // as a result, repeated model loading (e.g. from multiple calls of this chain) will increase VM usage
+    // although not fully resolving the problem, making the model loading a singleton can mitigate this problme
+
+    class CmrRealTimeLAXCineAIModel
+    {
+    public:
+
+        static void get_model_singleton(const std::string& model_home, const std::string& model_file, boost::python::object& model);
```

Comment:
Why does this function not just return the model (instead of writing it into an argument)?

Also, I think to make the pattern more recognizable, I think you should have. `instance()` just return a pointer to the singleton and then `get_model(model_home, model_file)` be a public function on the class, so you can do:

```cpp
auto model =  CmrRealTimeLAXCineAIModel::instance()->get_model(home,name);
```

Reply:
fixed as suggested.